### ISSUE TITLE
Couchbase Rocket example

### ISSUE THREAD

**Issue by himanshumps** [couchbaselabs/couchbase-rs#86]
Do we have any rest api example using rocket or any other server with couchbase rust api. I need to do a quick poc and I don't want to start from scratch.

**Comment by daschl** [comment]
I've created one for you to get you started: https://github.com/daschl/rocket-demo/blob/master/src/main.rs

**Comment by himanshumps** [comment]
I tried building that on github using github workflow but is failing while compiling libcouchbase

Logs attached: 
[logs_3.zip](https://github.com/couchbaselabs/couchbase-rs/files/5218394/logs_3.zip)

I have forked the code at https://github.com/himanshumps/rocket-demo.git

**Comment by himanshumps** [comment]
See the file build/3_Build.txt in attached log file

**Comment by daschl** [comment]
@himanshumps the error message provides a clue, you are missing environmental stuff:

```
 libev or libevent development files missing.
```

**Comment by himanshumps** [comment]
This is the log with RUST_BACKTRACE=1


[backtrace_enabled.txt](https://github.com/couchbaselabs/couchbase-rs/files/5218504/backtrace_enabled.txt)

**Comment by himanshumps** [comment]
I did not see your last post. Let me go through that

**Comment by daschl** [comment]
@himanshumps btw I don't know what you are after (and happy to chat via email if needed), but if you want higher performance the code I gave you blocks (only master of rocket is async aware). An alternative would be using actix-web or something lower level like hyper.

**Comment by himanshumps** [comment]
We are a vertx shop and we are seeing that we are not able to scale very well with the couchbase and vertx using async programming. I opened a issue in the couchbase forum about it (https://forums.couchbase.com/t/vert-x-couchbase-sdk-performance/27434/9) but we are stuck with that. I thought may be going the C way might improve performance and hence landed up trying to do the same stuff using rust. We are open to use actix-web or any other framework. We are looking for getting more than 15000 tps with 600 concurrent users (with may be 5-6 pods of 1 core CPU and 4GB RAM) on a 3-node couchbase cluster with each node having 8 core CPU and 64GB RAM . As we increase the number of users there is a decrease in throughput with vertx and couchbase. 

BTW I went through the C documentation and still the rust build is failing. I am doing the build on github using actions on ubuntu 18.04. Attached are the logs for the same. The github action file is at https://github.com/himanshumps/rocket-demo/tree/master/.github/workflows

[logs_11.zip](https://github.com/couchbaselabs/couchbase-rs/files/5219062/logs_11.zip)


Let me know what channel you want to use for chat. I am on couchbase forum as well with handle as himanshu.mps
